examples/03_connectivity/plot_confounds_removal.py

```diff
@@ -0,0 +1,199 @@
+
+References
+----------
+[1] Friston, K. J. et al. (1996).
+    Movement-related effects in fMRI time-series. Magnetic Resonance in
+    Medicine, 35(3), 346-355.
+
+[2] Behzadi, Y. et al. (2007).
+    A Component Based Noise Correction Method (CompCor) for BOLD and Perfusion
+    Based fMRI. Neuroimage, 37(1), 90-101.
+"""
+######################################
+# Getting the data
+# -----------------------------------------
+# We use single subject data from nilearn provided ADHD datasets.
+from nilearn import datasets
+adhd = datasets.fetch_adhd(n_subjects=1)
+func_filename = adhd.func[0]
+
+##########################################################################
+# Data include the functional image and a csv file of possible confounds.
+print(adhd.func)
+print(adhd.confounds)
+
+######################################
+# Friston et al. 24 motion parameters model
+# -----------------------------------------
+# They include the 6 motion parameters of the current volume and the preceding
+# volume, plus each of these values squared.
+
+# Read the 6 motion parameters from the confounds file
+import numpy as np
+csv_confounds = np.recfromcsv(adhd.confounds[0], delimiter='\t',
+                              usecols=('motionx', 'motiony', 'motionz',
+                                       'motionpitch', 'motionroll',
+                                       'motionyaw'))
+current_volume_motion = csv_confounds.view(dtype=float).reshape(-1, 6)
+preceding_volume_motion = np.vstack((current_volume_motion[0],
+                                     current_volume_motion[:-1]))
+confounds = np.hstack((current_volume_motion, preceding_volume_motion,
+                       current_volume_motion ** 2,
+                       preceding_volume_motion ** 2))
+
+##########################################################################
+# Anatomical CompCor
+# ----------------------------------------------
+# Signals within noisy ROIs, such as WM and CSF, are assumed to reflect noise.
+# The anatomical CompCor approach removes the first principal components
+# of noisy ROIs.
+#
+# First, we define the tissue masks.
+# As we do not have an anatomical image, we are relying on group-level masks.
+# Note that subject-specific masks would give more accurate results.
+
+##########################################################################
+# We give the paths to probabilistic group masks.
+icbm152 = datasets.fetch_icbm152_2009()
+mask_images = {'gm': icbm152.gm, 'wm': icbm152.wm, 'csf': icbm152.csf}
+
+##########################################################################
+# We use **nilearn.image.math_img** to convert probabilistic masks to binary
+# masks with thresholds being adjusted manually.
+from nilearn import image
+for tissue_name, threshold in zip(['wm', 'gm', 'csf'], ['.9', '.5', '.3']):
+    mask_images[tissue_name] = image.math_img('img > ' + threshold,
+                                              img=mask_images[tissue_name])
```

Comment:
I think that we have here another usecase for the `binarize_img` feature requested by @banilo